src/compiler/scala/tools/nsc/backend/jvm/BTypesFromSymbols.scala

```diff
@@ -108,8 +108,6 @@ class BTypesFromSymbols[G <: Global](val global: G) extends BTypes {
 
     assert(classSym != NoSymbol, "Cannot create ClassBType from NoSymbol")
     assert(classSym.isClass, s"Cannot create ClassBType from non-class symbol $classSym")
-    assertClassNotArrayNotPrimitive(classSym)
-    assert(!primitiveTypeToBType.contains(classSym) || isCompilingPrimitive, s"Cannot create ClassBType for primitive class symbol $classSym")
```

Comment:
I'm a bit sad to see those go.. They were helpful for finding bugs, and they are also useful documentation. Can we put them under a flag or elide them? Or even leave them as comments.